### PR TITLE
[New Rule] GCP Firewall Rule Deleted

### LINKED ISSUE #254: [New Rule] GCP Firewall Rule Deleted
## Description

Identifies when a firewall rule is deleted in Google Cloud Platform (GCP). Virtual Private Cloud (VPC) firewall rules can be configured to allow or deny connections to or from virtual machine (VM) instances. An adversary may delete a firewall rule in order to weaken their target's security controls.

```
event.module:googlecloud and event.dataset:googlecloud.audit and event.action:v*.compute.firewalls.delete
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Target Operating Systems:**

- **Platforms**

GCP

- **Target ECS Version:** `1.6.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
  - https://cloud.google.com/vpc/docs/firewalls

## Example Data

I found it interesting that the `event.outcome` is `unknown` when creating, modifying, or deleting firewall rules.

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #254 

## Summary
Identifies when a firewall rule is deleted in Google Cloud Platform (GCP). Virtual Private Cloud (VPC) firewall rules can be configured to allow or deny connections to or from virtual machine (VM) instances. An adversary may delete a firewall rule in order to weaken their target's security controls.

I'd like to include the [Impair Defenses](https://attack.mitre.org/techniques/T1562/) ATT&CK technique in this rule, but I think our schema might need to be updated before I can do this.

I found it interesting that the `event.outcome` is `unknown` when creating, modifying, or deleting firewall rules. I did not see a `success` value like many other GCP events. I decided to leave `event.outcome` out of this rule.

image

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/gcp/defense_evasion_gcp_firewall_rule_deleted.toml
Comment (threat-punter) [inline rules/gcp/defense_evasion_gcp_firewall_rule_deleted.toml:4]:
```suggestion
maturity = "production"
```

**Comment by threat-punter** [comment]
Blocked until I can add ATT&CK technique T1562 to the rule. cc @brokensound77

**Review by brokensound77 — APPROVED** [review]
LGTM 👍 

Should be unblocked once #330 merges

**Comment by threat-punter** [comment]
Worked like a charm. Cheers @brokensound77

File: rules/gcp/defense_evasion_gcp_firewall_rule_deleted.toml
Comment (bm11100) [inline rules/gcp/defense_evasion_gcp_firewall_rule_deleted.toml:23]:
```suggestion
name = "GCP Firewall Rule Deletion"
```

**Review by bm11100 — APPROVED** [review]
Just one suggestion of the name, otherwise, LGTM 🥇

**Pull request merged**
